zerver/middleware.py

```diff
@@ -576,16 +576,19 @@ def process_request(self, request: HttpRequest) -> Optional[HttpResponse]:
             return None
 
         subdomain = get_subdomain(request)
-        if (
-            subdomain != Realm.SUBDOMAIN_FOR_ROOT_DOMAIN
-            and subdomain != settings.SOCIAL_AUTH_SUBDOMAIN
-        ):
+        if subdomain != settings.SOCIAL_AUTH_SUBDOMAIN:
             request_notes = RequestNotes.get_notes(request)
             try:
                 request_notes.realm = get_realm(subdomain)
+                request_notes.has_fetched_realm = True
             except Realm.DoesNotExist:
-                return render(request, "zerver/invalid_realm.html", status=404)
-            request_notes.has_fetched_realm = True
+                # Requests to the root subdomain are still allowed even if there's
+                # no realm on it - it is used for creating new organizations
+                if not subdomain == Realm.SUBDOMAIN_FOR_ROOT_DOMAIN:
+                    return render(request, "zerver/invalid_realm.html", status=404)
+                else:
+                    pass
```

Comment:
This might be more readable as `if subdomain == Realm.SUBDOMAIN_FOR_ROOT_DOMAIN: return` with a comment, and then we can just return the 404 if that doesn't happen, avoiding the extra if/else layering.